test_rendering/spec/ol/layer/image.test.js

```diff
@@ -0,0 +1,98 @@
+goog.provide('ol.test.rendering.layer.Image');
+
+describe('ol.rendering.layer.Image', function() {
+
+  var target, map;
+
+  function createMap(renderer) {
+    target = createMapDiv(50, 50);
+
+    map = new ol.Map({
+      target: target,
+      renderer: renderer,
+      view: new ol.View({
+        center: ol.proj.transform(
+            [-122.416667, 37.783333], 'EPSG:4326', 'EPSG:3857'),
+        zoom: 5
+      })
+    });
+    return map;
+  }
+
+  function waitForImages(sources, layerOptions, onImagesLoaded) {
+    var imagesLoading = 0;
+    var imagesLoaded = 0;
+
+    var update = function() {
+      if (imagesLoading === imagesLoaded) {
+        onImagesLoaded();
+      }
+    };
+
+    goog.array.forEach(sources, function(source) {
+      source.on('imageloadstart', function(event) {
+        console.log('start ' + imagesLoading + ' ' + imagesLoaded);
+        imagesLoading++;
+      });
+      source.on('imageloadend', function(event) {
+        console.log('end ' + imagesLoading + ' ' + imagesLoaded);
+        imagesLoaded++;
+        update();
+      });
+      source.on('imageloaderror', function(event) {
+        console.log('error ' + imagesLoading + ' ' + imagesLoaded);
```

Comment:
Would it make sense to call `onImagesLoaded` with an `Error` here?  Then the tests could call `done` with the same.  Then the `console.log` calls could probably be removed.